adagrams/game.py

```diff
@@ -1,11 +1,74 @@
-def draw_letters():
-    pass
+import random
+letter_set1 = ["A","I"]* 9
+letter_set2 = ["B","C","F","H","M","P","V","W","Y"] * 2
+letter_set3 = ["D","L","S","U"] * 4
+letter_set4 = ["E"] * 12
+letter_set5 = ["G"] * 3
+letter_set6 = ["J","K","Q","X","Z"]
+letter_set7 = ["N","R","T"] * 6
+letter_set8 = ["O"]* 8
+letters_list = letter_set1 + letter_set2 + letter_set3 + letter_set4\
+               + letter_set5 + letter_set6 + letter_set7 + letter_set8
+letter_score_dict ={}
+for letter in ["A", "E", "I", "O", "U","L", "N", "R","S","T"]:
+    letter_score_dict[letter] = 1
+dict_items = {"D":2,"G":2,"K":5,"J":8,"X":8,"Q":10,"Z":10}
+letter_score_dict.update(dict_items)
+for letter in ["B", "C", "M", "P"]:
+    letter_score_dict[letter]= 3
+for letter in ["F", "H", "V", "W", "Y"]:
+    letter_score_dict[letter]= 4
 
+def draw_letters():
+   
+   letter_hand = random.sample(letters_list, 10)
+   return letter_hand
+   
 def uses_available_letters(word, letter_bank):
-    pass
+   flag = True
+   letter_bank_copy = letter_bank.copy()
+   
+   for char in word:
+       
+       if char.upper() in letter_bank_copy:
+           
+           letter_bank_copy.pop(letter_bank_copy.index(char.upper()))
+           
+       else:
+           flag = False
+       
+
 
+   return flag
+           
+           
 def score_word(word):
-    pass
+    score = 0
+    for char in word:
+        score += letter_score_dict[char.upper()]
+    if len(word) >= 7:
+        score += 8
+
+    return score
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    word_scores = []
+    for word in word_list:
+        score = score_word(word)
+        word_scores.append(score)
+    best_score = max(word_scores)
+    for word in word_list:
+        if len(word) == 10 and score_word(word) == best_score:
```

Comment:
Nice job coming up with this approach for breaking ties!

Notice that you call `score_word(word)` on line 57 and then again on line 61. How can we avoid this repetition? While it is a just a small piece of code repeated, we should try to keep our code as DRY as possible, especially since score_word is a method that has a for loop in it so we call a method that executes a for loop 2 times. 

While we haven't covered Big O yet, we'll soon learn that when we have method that has a loop in it (like score_word) and then we call that method inside a loop (line 60), we'll negatively affect the performance of code. 

How could you rework your solution so that you don't need to call score_word? Could you create a word_scores dictionary instead of a list so that you can keep track of a word and its score? Like {"AAAAAAAAAA": 18, "EEEEEEEEEE": 18}